scripts/core-ng-module-generator.js

```diff
@@ -0,0 +1,172 @@
+const indexTemplate = fs.readFileSync(MUSTACHE_TEMPLTATE_PATH + '/index.ts.mustache', 'utf8');
+const rootIndexTemplate = fs.readFileSync(MUSTACHE_TEMPLTATE_PATH + '/root-index.ts.mustache', 'utf8');
+const rootModuleTemplate = fs.readFileSync(MUSTACHE_TEMPLTATE_PATH + '/root-module.ts.mustache', 'utf8');
+const testTemplate = fs.readFileSync(MUSTACHE_TEMPLTATE_PATH + '/test.ts.mustache', 'utf8');
+
+function getDirectiveClassName(tagName, prefix, suffix) {
+  return (
+    prefix +
+    tagName
+      .split('-')
+      .map(w => w.charAt(0).toUpperCase() + w.slice(1))
+      .join('') +
+    suffix
+  );
+}
+
+function getDiretoryName(tagPath) {
+  return tagPath.split('/')[1];
+}
+
+function getDirectiveName(tagName) {
+  return tagName + '.directive.ts';
+}
+
+function createFileOnTemplate(template, templateData, name, targetPath, subDir = '') {
+  if (fs.existsSync(path.join(targetPath, subDir))) {
+    const filePath = path.join(targetPath, subDir, name);
+    const templateRender = Mustache.render(template, templateData);
+    fs.writeFileSync(filePath, templateRender);
+  }
+}
+
+function createRootIndex(moduleData) {
+  if (moduleData) {
+    const templateData = {
+      modules: Object.keys(moduleData).map(md => ({ moduleDirectoryName: md })),
+    };
+
+    createFileOnTemplate(rootIndexTemplate, templateData, 'index.ts', TARGET_PATH);
+  }
+}
+
+function createIndexes(moduleData) {
+  if (moduleData) {
+    Object.keys(moduleData).forEach(moduleDirectory => {
+      const templateData = {
+        directives: Object.keys(moduleData[moduleDirectory]).map(md => ({
+          directiveFileName: md,
+        })),
+        moduleFileName: 'cds-' + moduleDirectory,
+      };
+
+      createFileOnTemplate(indexTemplate, templateData, 'index.ts', TARGET_PATH, moduleDirectory);
+    });
+
+    createRootIndex(moduleData);
+  }
+}
+
+function createModules(moduleData) {
+  if (moduleData) {
+    const rootModuleTemplateData = { modules: [] };
+    Object.keys(moduleData).forEach(moduleDirectory => {
+      const templateData = {
+        directives: Object.keys(moduleData[moduleDirectory]).map(md => ({
+          directiveFileName: md,
+          directiveClassName: moduleData[moduleDirectory][md],
+        })),
+        moduleClassName: getDirectiveClassName(moduleDirectory, 'Cds', 'Module'),
+      };
+
+      rootModuleTemplateData.modules.push({ moduleClassName: templateData.moduleClassName, moduleDirectory });
+
+      const moduleFileName = 'cds-' + moduleDirectory + '.module.ts';
+      createFileOnTemplate(moduleTemplate, templateData, moduleFileName, TARGET_PATH, moduleDirectory);
+    });
+
+    createFileOnTemplate(rootModuleTemplate, rootModuleTemplateData, 'cds.module.ts', TARGET_PATH);
+
+    return moduleData;
+  }
+}
+
+function createDirectives(componentDirectories) {
+  if (componentDirectories) {
+    const customElementsData = JSON.parse(fs.readFileSync(SOURCE_PATH));
```

Comment:
`JSON.parse` has the tendency to throw SyntaxError when something gets even close to wrong. Is it better to have `try/catch`